vignettes/netCDF_Functionality.Rmd

```diff
@@ -0,0 +1,400 @@
+
+```{r, TimeAtt, results = 'hide'}
+time_attributes <- list(
+  name = 'time',
+  units = 'days since 1900-01-01',
+  calendar = 'standard',
+  unlim = TRUE,
+  vals = time_midpoints # mid point of year
+  )
+```
+
+### Vertical (depth / Z) Attributes
+
+A named list describing the vertical (depth / Z) dimension. The *name*, *units*,
+and *vals* fields are required.
+
+```{r, VertAtt, results = 'hide'}
+vertical_attributes <- list(
+  name = 'depth',
+  units = 'cm',
+  positive = 'down', # depth is down, height is up
+  vals = depths_max
+)
+```
+
+### Global Attributes
+
+A named list describing the global attributes of the netCDF. This is where you describe
+things like how the data was made, where the data was made, who the data was made by,
+what climate models were used to create the data, etc.
+
+The *title*, *institution*, *description*, *source_id*, *realm*, *product*, and
+*contact* fields are required.
+
+```{r, GlobalAtt, results = 'hide'}
+global_attributes <- list(
+    title = "Project Name Annual Wet Degree Days",
+    institution = 'Southwest Biological Science Center, U.S. Geological Survey',
+    description = 'methods on how this data was made',
+    source = paste(
+      "SOILWAT2 (v4.2.0);",
+      "rSOILWAT2 (v2.3.2);",
+      "rSFSW2 (v3.1.2)"
+    ),
+    source_id = "SOILWAT2",
+    realm = "land",
+    parent_mip_era = "CMIP5",
+    parent_experiment_id = "RCP45",
+    parent_source = "CanESM2",
+    parent_variant_label = "r1i1p1f1",
+    product = "model-output",
+    projection = 'Geographic',
+    grid = 'WGS84',
+    grid_label = "gn",
+    nominal_resolution = "10 km", # \code{\link{populate_netcdf_from_array}}
+    further_info_url = "https://github.com/DrylandEcology/",
+    contact = "you@email.com"
+    )
+```
+
+#### Nominal resolution
+
+If the data is calculated and reported on a grid, CF convention calls for the
+field *nominal resolution* to be contained in the global attributes. The function
+*calculate_nominal_resolution* will calculate the nominal resolution of your gridded
+data for you. However, you need to provide a raster of the grid, the lat/long locations, and
+the area that each cell in the grid represents.
+
+```{r, Nominal, results = 'hide'}
+r <- raster::raster(
+  xmn = -120, xmx = -90,
+  ymn = 30, ymx = 50,
+  crs ="+init=epsg:4326",
+  resolution = c(0.5, 0.5)
+)
+r[] <- seq_len(prod(dim(r)))
+xy <- raster::sampleRandom(r, size = 50, sp = TRUE)
+grid_cell_area <- calculate_cell_area(sites = xy, grid = r)
+
+calculate_nominal_resolution(
+  grid = r,
+  sites = xy,
+  cell_areas_km2 = grid_cell_area[, "km2"]
+)
+```
+
+# Output Files Structure Naming Conventions
+
+File structure should be organized by: variableName / allFilesPerVariable. Each file
+will contain variable output for one time period and/or climate model.
+
+The name for each file should be in the format as follows:
+**var_LorA_timestep_Project_SOILWAT2_GCM_RCP_variantLabel_gn_startT_endT.nc**
```

Comment:
CMIP uses startT-endT instead

Reply:
Should we also add a `-clim` suffix if the file contains climatologies instead of time-series?